src/Hyperion.Tests/UnsafeDeserializationExclusionTests.cs

```diff
@@ -25,6 +34,73 @@ public void CantDeserializeANaughtyType()
             }
         }
 
+        [Theory]
+        [MemberData(nameof(DangerousObjectFactory))]
+        public void DetectNaughtyTypesByDefault(Type dangerousType)
```

Comment:
Detection hardening, actually try to use the dangerous types and see if they are being caught by the default filter